src/graph/sampler.cc

```diff
@@ -1042,10 +1161,16 @@ DGL_REGISTER_GLOBAL("sampling._CAPI_UniformEdgeSampling")
 
       Subgraph subg = gptr->EdgeSubgraph(worker_seeds, false);
       positive_subgs[i] = ConvertRef(subg);
-      if (neg_mode.size() > 0) {
-        Subgraph neg_subg = NegEdgeSubgraph(gptr->NumVertices(), subg,
-                                            neg_mode, neg_sample_size,
-                                            gptr->IsMultigraph(), exclude_positive);
+      // For PBG negative sampling, we accept "PBG-head" for corrupting head
+      // nodes and "PBG-tail" for corrupting tail nodes.
+      if (neg_mode.substr(0, 3) == "PBG") {
+        Subgraph neg_subg = PBGNegEdgeSubgraph(gptr->NumVertices(), subg,
+                                               neg_mode.substr(4), neg_sample_size,
```

Comment:
In the PGB paper, the sampler do both head corrupt and tail corrupt together.
Here it seems the python code should call the PBGhead and PBGtail

Reply:
There are two modes "PBG-head" and "PBG-tail". A user needs to call this method twice to corrupt head nodes and tail nodes. This is what we do for regular negative sampler.